### PR TITLE
Ctt fixes

### PR DESCRIPTION
**Pull request by opcua-tsn-team-kalycito**
Contains fixes for CTT

### CODE REVIEW

File: src/server/ua_server_binary.c [hunk 1]
```diff
@@ -519,8 +519,9 @@ processMSG(UA_Server *server, UA_SecureChannel *channel,
         session = &anonymousSession;
     }
 
-    /* Trying to use a non-activated session? */
-    if(sessionRequired && !session->activated) {
+    /* Trying to use a non-activated session?
+     * Do not allow if request is of type CloseSessionRequest */
+    if(sessionRequired && !session->activated && strcmp(requestType->typeName, "CloseSessionRequest")) {
```
Comment (jpfr) [inline src/server/ua_server_binary.c]: I'm not sure this has the intended effect.
This allows access if
- The service requires a session
- The session is not activated
- The service is **not** a CloseSessionRequest
Please close this pontential security hole.

Can you move the test for session->activated here?
https://github.com/open62541/open62541/blob/4209f74e2176f75f67b4f80962909c69cdb54126/src/server/ua_services_session.c#L368

Comment (jpfr) [inline src/server/ua_server_binary.c]: The strcmp is very indirect logic. Also, there is a build-flag to disable the typenames.
To save ROM on embedded targets. Another solution is
```c
if(sessionRequired && !session->activated && requestType != &UA_TYPES[UA_TYPES_CLOSESESSIONREQUEST])
```
That's also a bit faster.

File: src/server/ua_services_session.c [hunk 2]
```diff
@@ -213,6 +213,11 @@ checkSignature(const UA_Server *server, const UA_SecureChannel *channel,
        channel->securityMode != UA_MESSAGESECURITYMODE_SIGNANDENCRYPT)
         return UA_STATUSCODE_GOOD;
 
+    /* Check for zero signature length in client signature */
+    if(!(request->clientSignature.signature.length)) {
```
Comment (jpfr) [inline src/server/ua_services_session.c]: Can you make this an explicit test?
```c
if(request->clientSignature.signature.length == 0)
    return UA_STATUSCODE_BADAPPLICATIONSIGNATUREINVALID;
```

Comment (opcua-tsn-team-kalycito) [reply src/server/ua_services_session.c]: Done!

File: src/server/ua_server_binary.c (hunk 1 above)
Comment (opcua-tsn-team-kalycito) [reply src/server/ua_server_binary.c]: Done!

**Comment by jpfr** [comment]
Merged. Thanks!

**Pull request merged**